crates/cast/src/script.rs

```diff
@@ -0,0 +1,360 @@
+    pub runtime: Runtime,
+}
+
+// crates/blockifier/src/execution/syscalls/hint_processor.rs:472 (ResourceTracker for SyscallHintProcessor)
+impl ResourceTracker for CairoHintProcessor<'_> {
+    fn consumed(&self) -> bool {
+        self.blockifier_syscall_handler
+            .context
+            .vm_run_resources
+            .consumed()
+    }
+
+    fn consume_step(&mut self) {
+        self.blockifier_syscall_handler
+            .context
+            .vm_run_resources
+            .consume_step();
+    }
+
+    fn get_n_steps(&self) -> Option<usize> {
+        self.blockifier_syscall_handler
+            .context
+            .vm_run_resources
+            .get_n_steps()
+    }
+
+    fn run_resources(&self) -> &RunResources {
+        self.blockifier_syscall_handler
+            .context
+            .vm_run_resources
+            .run_resources()
+    }
+}
+
+impl HintProcessorLogic for CairoHintProcessor<'_> {
+    fn execute_hint(
+        &mut self,
+        vm: &mut VirtualMachine,
+        exec_scopes: &mut ExecutionScopes,
+        hint_data: &Box<dyn Any>,
+        constants: &HashMap<String, Felt252>,
+    ) -> Result<(), HintError> {
+        let maybe_extended_hint = hint_data.downcast_ref::<Hint>();
+        if let Some(Hint::Starknet(StarknetHint::Cheatcode {
+            selector,
+            input_start,
+            input_end,
+            output_start,
+            output_end,
+        })) = maybe_extended_hint
+        {
+            return self.execute_cheatcode_hint(
+                vm,
+                exec_scopes,
+                selector,
+                input_start,
+                input_end,
+                output_start,
+                output_end,
+            );
+        }
+        self.blockifier_syscall_handler
+            .execute_hint(vm, exec_scopes, hint_data, constants)
+    }
+
+    /// Trait function to store hint in the hint processor by string.
+    fn compile_hint(
+        &self,
+        hint_code: &str,
+        _ap_tracking_data: &ApTracking,
+        _reference_ids: &HashMap<String, usize>,
+        _references: &[HintReference],
+    ) -> Result<Box<dyn Any>, VirtualMachineError> {
+        Ok(Box::new(self.hints[hint_code].clone()))
+    }
+}
+
+impl CairoHintProcessor<'_> {
+    #[allow(clippy::trivially_copy_pass_by_ref, clippy::too_many_arguments)]
+    pub fn execute_cheatcode_hint(
+        &mut self,
+        vm: &mut VirtualMachine,
+        _exec_scopes: &mut ExecutionScopes,
+        selector: &BigIntAsHex,
+        input_start: &ResOperand,
+        input_end: &ResOperand,
+        output_start: &CellRef,
+        output_end: &CellRef,
+    ) -> Result<(), HintError> {
+        // Parse the selector.
+        let selector = &selector.value.to_bytes_be().1;
+        let selector = std::str::from_utf8(selector).map_err(|_| {
+            HintError::CustomHint(Box::from(
+                "Failed to parse the  cheatcode selector".to_string(),
```

Comment:
In some cases we return hinterror and in some other we use `expect/unwrap`. This is a bit inconsistent.

Reply:
If you mean this https://github.com/foundry-rs/starknet-foundry/pull/1067/files#diff-8355c5a63f2b3a250d96b5722de9db86497c0946b175a45c11506f9626d14724R196-R227 piece of logic, it is consistent with the way `TestExecutionSyscallHandler` handles it.

Reply:
Sure but the logic there is not ideal, so we can consider not copying it in the exact manner.

Reply:
ok, let's consider reworking it in a separate PR

https://github.com/foundry-rs/starknet-foundry/issues/1168